### PR TITLE
Fix section teleport bugs with entities, use new style guide for all files

### PR DESCRIPTION
**Pull request by Norazan**
In hindsight I should have probably made the fixes and style guide change two seperate commits.

The main code changes are in DeeperWorld.kt, Tree.kt and MovementListener.kt.

### CODE REVIEW

File: src/main/java/com/derongan/minecraft/deeperworld/extensions/DeeperWorldExtensions.kt [hunk 1]
```diff
@@ -27,7 +27,11 @@ internal fun Entity.getPassengersRecursive(): List<Entity> {
 }
 
 internal fun Player.getLeashedEntities(): List<LivingEntity> {
-    return getNearbyEntities(20.0, 20.0, 20.0) // Max leashed entity range is 10 blocks, therefore these parameter values
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/extensions/DeeperWorldExtensions.kt:30]: I think put the comment on the line above, then the auto formatter won't try so aggressively to make it all fit on one line, it looked cleaner before.

File: src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt [hunk 2]
```diff
@@ -50,11 +54,21 @@ object MovementListener : Listener {
...
             if (!DeeperConfig.data.damageExcludedWorlds.contains(player.location.world)
-                    && DeeperConfig.data.damageOutsideSections > 0.0
-                    && (player.gameMode == GameMode.SURVIVAL || player.gameMode == GameMode.ADVENTURE)) {
+                && DeeperConfig.data.damageOutsideSections > 0.0
+                && (player.gameMode == GameMode.SURVIVAL || player.gameMode == GameMode.ADVENTURE)
+            ) {
                 player.damage(0.01) //give a damage effect
-                player.health = (player.health - DeeperConfig.data.damageOutsideSections / 10).coerceIn(0.0, player.getAttribute(Attribute.GENERIC_MAX_HEALTH)?.value) //ignores armor
-                player.sendTitle("&cYou are not in a managed section".color(), "&7You will take damage upon moving!".color(), 0, 20, 10)
+                player.health = (player.health - DeeperConfig.data.damageOutsideSections / 10).coerceIn(
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt:61]: Put .coerceIn on the next line so it only takes up 2 lines

File: src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt [hunk 3]
```diff
@@ -50,11 +54,21 @@ object MovementListener : Listener {
...
+                    0.0,
+                    player.getAttribute(Attribute.GENERIC_MAX_HEALTH)?.value
+                ) //ignores armor
+                player.sendTitle(
+                    "&cYou are not in a managed section".color(),
+                    "&7You will take damage upon moving!".color(),
+                    0,
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt:68]: Keep the three numbers on same line

File: src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt [hunk 4]
```diff
@@ -132,44 +162,106 @@ object MovementListener : Listener {
...
             }
 
-            player.teleport(newLoc)
-
-            deeperWorld.schedule {
-                waitFor(DeeperConfig.data.entityTeleportDelay)
+            // If the vehicleTree contains a minecart, teleporting the player across worlds within the same tick
+            // as removing the passenger throws a "Removing ticking entity!" exception.
+            // Delay the teleportation by 1 tick in this case.
+            if (vehicleTree.root.where { it.value.type == EntityType.MINECART }.isNotEmpty()) {
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt:168]: Are there any issues with doing this regardless of whether there's a minecart?

**Review by 0ffz — CHANGES_REQUESTED** [review]
A couple formatting change requests and quick question for minecarts

File: src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt (hunk 4 above)
Comment (Norazan) [reply src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt:168]: Nope, should be fine. I guess we can always wait 1 tick after the passenger removal before teleporting.

**Review by 0ffz — APPROVED** [review]

**Pull request merged**